### PR TITLE
docs: add guides for standalone APIs in module-based apps

### PR DESCRIPTION
**Pull request by KylerJohnsonDev**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

For those working in module-based angular apps who want to leverage standalone components going forward, it's a bit confusing when you register feature state in your standalone components and receive a Null Injector error for the root store. This adds guides for registering root store and root effects for use in standalone components in module-based Angular applications. 

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Note, we may move this to its own guide, but this is a good start

File: projects/ngrx.io/content/guide/effects/index.md [hunk 1]
```diff
@@ -346,6 +346,32 @@ The `EffectsModule.forFeature()` method or `provideEffects()` function must be a
...
+
+import { MoviesEffects } from './effects/movies.effects';
+import * as actorsEffects from './effects/actors.effects';
+
+@NgModule({
+  imports: [
+    EffectsModule.forRoot(MoviesEffects, actorsEffects),
+  ],
+  providers: [
+    provideEffects(MoviesEffects, actorsEffects)
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/effects/index.md:367]: Can you add `provideEffects` to the imports please ?

File: projects/ngrx.io/content/guide/store/reducers.md [hunk 2]
```diff
@@ -272,6 +272,30 @@ After the feature is loaded, the `game` key becomes a property in the object and
...
+  import { NgModule } from '@angular/core';
+  import { StoreModule } from '@ngrx/store';
+  import { scoreboardReducer } from './reducers/scoreboard.reducer';
+
+  @NgModule({
+    imports: [
+      StoreModule.forRoot({ game: scoreboardReducer })
+    ],
+    providers: [
+      provideStore({ game: scoreboardReducer })
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/reducers.md:292]: Can you add `provideStore` to the imports please ?

**Review by timdeschryver — COMMENTED** [review]
What's you opinion about moving this to the "Registering root state" section? My reasoning is that this only applies to root state/effects, feel free to correct me if I'm wrong about this 😅

image

**Comment by KylerJohnsonDev** [comment]
@timdeschryver It may make more sense to put it there. But for context, I ran into this as an issue when trying to use feature state. I added the provider for the feature state to the route config and when I injected the store into the component, it said there was no provider for root state. Of course, that's because I wasn't using `provideStore` in "AppModule". Root state was only registered using `StoreModule.forRoot({...})`.

File: projects/ngrx.io/content/guide/effects/index.md (hunk 1 above)
Comment (KylerJohnsonDev) [reply projects/ngrx.io/content/guide/effects/index.md:367]: I'm a bit confused by this request. Under the existing stand-alone section, it shows adding `provideEffects` as a provider. So I did the same here for consistency.

File: projects/ngrx.io/content/guide/store/reducers.md (hunk 2 above)
Comment (KylerJohnsonDev) [reply projects/ngrx.io/content/guide/store/reducers.md:292]: I'm a similarly confused by this request, too. Under the existing stand-alone section, it shows adding provideStore as a provider. So I did the same here for consistency.

File: projects/ngrx.io/content/guide/effects/index.md (hunk 1 above)
Comment (timdeschryver) [reply projects/ngrx.io/content/guide/effects/index.md:367]: Oh yeah, this is confusing, my bad 😅
I meant to add an import statement add the top of the page:

```ts
import { provideEffects } from '@ngrx/effects';
```

File: projects/ngrx.io/content/guide/store/reducers.md (hunk 2 above)
Comment (timdeschryver) [reply projects/ngrx.io/content/guide/store/reducers.md:292]: Same here but with

```ts
import { provideStore } from '@ngrx/store';
```

**Comment by timdeschryver** [comment]
@KylerJohnsonDev that's a fair point, thanks for the elaboration.
Let's keep it like this 👍

**Review by timdeschryver — APPROVED** [review]
Thanks!

**Pull request merged**